pkg/controllers/workgenerator/controller.go

```diff
@@ -792,21 +794,21 @@ func (r *Reconciler) upsertWork(ctx context.Context, newWork, existingWork *flee
 // The corresponding work name prefix is the CRP name + sub-index if there is a sub-index. Otherwise, it is the CRP name +"-work".
 // For example, if the resource snapshot name is "crp-1-0", the corresponding work name is "crp-0".
 // If the resource snapshot name is "crp-1", the corresponding work name is "crp-work".
-func getWorkNamePrefixFromSnapshotName(resourceSnapshot *fleetv1beta1.ClusterResourceSnapshot) (string, error) {
+func getWorkNamePrefixFromSnapshotName(resourceSnapshot fleetv1beta1.ResourceSnapshotObj) (string, error) {
 	// The validation webhook should make sure the label and annotation are valid on all resource snapshot.
 	// We are just being defensive here.
-	crpName, exist := resourceSnapshot.Labels[fleetv1beta1.CRPTrackingLabel]
+	crpName, exist := resourceSnapshot.GetLabels()[fleetv1beta1.CRPTrackingLabel]
```

Comment:
Hi Ryan! The `crpName` name part might be a bit confusing IMO. Of course for now it is revelant...

Reply:
I haven't got to this controller yet, this is just to fix the resourceSnapshot.